### PR TITLE
feat(manger/asdf): add istioctl and coockiecutter support

### PR DESCRIPTION
**Pull request by kamilaz**
## Changes

Added istioctl and cookiecutter asdf plugins to the asdf manager

## Context

#30887

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 1]
```diff
@@ -116,6 +116,14 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
       versioning: `${regexVersioning.id}:^(?<major>\\d+?)\\.(?<minor>\\d+?)\\.(?<patch>\\d+)\\.(?<build>\\d+)$`,
     },
   },
+  cookiecutter: {
+    asdfPluginUrl: 'https://github.com/shawon-crosen/asdf-cookiecutter',
+    config: {
+      datasource: GithubReleasesDatasource.id,
+      packageName: 'cookiecutter/cookiecutter',
```
Comment (viceice) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:123]: add semver versioning

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 2]
```diff
@@ -116,6 +116,14 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
...
+      extractVersion: '^v(?<version>\\S+)',
```
Comment (viceice) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:124]:
```suggestion
```
there is no leading v


https://github.com/cookiecutter/cookiecutter/releases

File: lib/modules/manager/asdf/upgradeable-tooling.ts [hunk 3]
```diff
@@ -302,6 +310,15 @@ export const upgradeableTooling: Record<string, ToolingDefinition> = {
       extractVersion: '^v(?<version>\\S+)',
     },
   },
+  istioctl: {
+    asdfPluginUrl: 'https://github.com/virtualstaticvoid/asdf-istioctl',
+    config: {
+      datasource: GithubReleasesDatasource.id,
+      packageName: 'istio/istio',
+      versioning: semverVersioning.id,
+      extractVersion: '^v(?<version>\\S+)',
```
Comment (viceice) [inline lib/modules/manager/asdf/upgradeable-tooling.ts:319]:
```suggestion
```

File: lib/modules/manager/asdf/upgradeable-tooling.ts (hunk 1 above)
Comment (kamilaz) [reply lib/modules/manager/asdf/upgradeable-tooling.ts:123]: added

File: lib/modules/manager/asdf/upgradeable-tooling.ts (hunk 2 above)
Comment (kamilaz) [reply lib/modules/manager/asdf/upgradeable-tooling.ts]: extractVersion removed

File: lib/modules/manager/asdf/upgradeable-tooling.ts (hunk 3 above)
Comment (kamilaz) [reply lib/modules/manager/asdf/upgradeable-tooling.ts]: extractVersion removed

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
don't merge main, we're using GitHub merge queue. please also resolve conversations instead of commenting when done

**Pull request merged**